core/src/main/com/sinch/sdk/core/http/HttpContentType.java

```diff
@@ -17,4 +23,20 @@ public static boolean isMimeJson(Collection<String> mimes) {
   public static boolean isMimeTextPlain(Collection<String> mimes) {
     return mimes.stream().anyMatch(TEXT_PLAIN::equalsIgnoreCase);
   }
+
+  public static Optional<String> getCharsetValue(String contentTypeHeader) {
+    if (StringUtil.isEmpty(contentTypeHeader)) {
+      return Optional.empty();
+    }
+
+    Matcher m = charsetPattern.matcher(contentTypeHeader);
+    if (!m.find()) {
+      return Optional.empty();
+    }
+    String group = m.group(2);
+    if (group.contains(";")) {
```

Comment:
You may update the regex to exclude the `;` and any space that may be present, such as `\\s*([^;\\s]+)`

Reply:
Nice !
Done